### PR TITLE
Load GPU trained model on cpu or vice versa

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

### CODE REVIEW

File: unionml/model.py [hunk 1]
```diff
@@ -594,7 +594,10 @@ def _default_loader(self, file: Union[str, os.PathLike, IO], *args, **kwargs) ->
             deserialized_model = joblib.load(file, *args, **kwargs)
             return deserialized_model["model_state"]
         elif issubclass(model_type, torch.nn.Module):
-            deserialized_model = torch.load(file, *args, **kwargs)
+            if not torch.cuda.is_available():
+                deserialized_model = torch.load(file, map_location=torch.device("cpu"), *args, **kwargs)
+            else:
+                deserialized_model = torch.load(file, map_location=torch.device("cuda"), *args, **kwargs)
```
Comment (cosmicBboy) [inline unionml/model.py:600]: why is this necessary? `map_location` can simply be passed as a kwarg when calling the loader.

**Review by cosmicBboy — CHANGES_REQUESTED** [review]

File: unionml/model.py (hunk 1 above)
Comment (kumare3) [reply unionml/model.py:600]: @cosmicBboy ignore this PR, a better fix is needed. Let's discuss Monday

**Comment by cosmicBboy** [comment]
gonna close this for now
